app.py

```diff
@@ -98,38 +101,45 @@
     registry=prometheus_registry,
 )
 # If set to non-zero value, the graph-refresh will be scheduled for only first N unsolved package-versions.
-_THOTH_GRAPH_REFRESH_EAGER_STOP = int(os.getenv("THOTH_GRAPH_REFRESH_EAGER_STOP") or 0)
+_THOTH_GRAPH_REFRESH_EAGER_STOP = int(os.getenv("THOTH_GRAPH_REFRESH_EAGER_STOP") or _GRAPH_DB.DEFAULT_COUNT)
 
 
 def graph_refresh_solver() -> None:
     """Schedule refresh for packages that are not yet analyzed by solver."""
+    _LOGGER.info(
+        "Eager stop of scheduling new solver runs for unsolved package versions, packages scheduled: %d",
+        _THOTH_GRAPH_REFRESH_EAGER_STOP,
+    )
+
     indexes = list(_GRAPH_DB.get_python_package_index_urls())
 
     packages = []
     # Iterate over all registered solvers and gather packages which were not solved by them.
     for solver_name in _OPENSHIFT.get_solver_names():
         _LOGGER.info("Checking unsolved packages for solver %r", solver_name)
-        for package_name, version in _GRAPH_DB.retrieve_unsolved_python_packages(solver_name):
-            _LOGGER.info(f"Adding new package %r in version %r", package_name, version)
+        solver_info = _GRAPH_DB.parse_python_solver_name(solver_name)
+        for package_name, version, index_url in _GRAPH_DB.get_unsolved_python_package_versions(
+            os_name=solver_info["os_name"],
+            os_version=solver_info["os_version"],
+            python_version=solver_info["python_version"],
+            count=_THOTH_GRAPH_REFRESH_EAGER_STOP
+        ):
+            _LOGGER.info(
+                f"Adding new package %r in version %r on index %r",
+                package_name,
+                version,
+                index_url,
+            )
             _METRIC_PACKAGES_ADDED.inc()
-            packages.append((package_name, version, solver_name))
+            packages.append((package_name, version, index_url, solver_name))
 
     if not packages:
         _LOGGER.info("No unsolved packages found")
         return
 
-    count = 0
-    for package_name, package_version, solver_name in packages:
+    for package_name, package_version, index_url, solver_name in packages:
+        indexes = [index_url] or indexes
```

Comment:
One minor thing spotted - the indexes variable as obtained from the database will be overwritten during iterations. To avoid it, you can directly do in the line below:

```python
for index_url in [index_url] if index_url is not None else indexes:
```

(Also, it will always default to True as `bool([None]) == True`).